.github/workflows/php.yml

```diff
@@ -0,0 +1,64 @@
+name: PHP Composer
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version:
+          - "8.0"
+          - "8.1"
+
+        include:
+          - os: ubuntu-latest
+            php-version: "8.0"
+            compiler: jit
+
+          - os: ubuntu-latest
+            php-version: "8.1"
+            compiler: jit
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install PHP with extensions
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ matrix.php-version }}
+        extensions: memcached
+        coverage: none
+
+    - name: Setup Memcached server
+      uses: niden/actions-memcached@v7
+
+    - name: Validate composer.json and composer.lock
+      run: composer validate --strict
+
+    - name: Cache Composer packages
```

Comment:
Mind explaining what this section does? Looks like we cache the composer lock file on a per `os` basis but it looks like only run on one `os`.

Reply:
I think the OS is in the cache key, so that it _doesn't_ get cached across different OS's.

I'm not sure if it caches just across different steps, or also subsequent CI runs? I stole this from the PHP CI template, and forgot to look into it more.

Reply:
https://github.com/actions/cache

I think we're caching the vendor directory (i.e., everything installed by composer), and only invalidating the cache if a composer.lock file (in any directory) changes (different hash), or if the OS changes. In the event of a hit, we restore the vendor directory from the cache. Caching is scoped to the current branch, but if that's a miss, it will try the parent branch, so if a branch doesn't change composer.lock, then I expect it to get the vendor dir from its parent (probably master). The `restore-keys` argument is used if there's a miss on `key`, and it's a most-to-least-specific list of key prefixes to try. So even if there isn't a hit for the current composer.lock, this config will pull the most recent cached value from the branch scope, falling back to the parent branch scope.

Basically, we're going to try really hard to get the vendor dir from _some_ previous build, and then we're going to run `composer install` on top of that. We have the option to skip a job based on a previous cache job resulting in a hit, but we wouldn't want that here because our key config means that we can get a vendor dir that doesn't fully satisfy our composer.lock.

There's a 10G per repo limit on total cache size, and items are evicted after 7 days without access.

Reply:
Thanks for always being thorough and informative Shawn!